# 1 Probleemstelling

Blackjack is een kaartspel wat veel gespeeld wordt in casino's. De speler speelt niet tegen andere spelers, maar tegen de bank. Het doel van Blackjack is om zo dicht bij de 21 punten te halen zonder er overheen te gaan. Als de speler dichterbij de 21 komt dan de bank, dan wint de speler. Als de bank dichterbij komt wint de bank (*Blackjack Spelregels - Regels Voor Blackjack | TOTO*, z.d.; *Blackjack: Uitleg & Spelregels*, 2025; Spelregels.Eu, 2020; Van Geest, 2014).

## 1.1 Regels van blackjack

### 1.1.1 De kaarten delen
Zodra de speler de inzet geplaatst heeft, worden de kaarten gedeeld. De speler krijgt twee zichtbare kaarten. De bank krijgt eveneens twee kaarten, waarvan er één zichtbaar is voor de speler (*Blackjack Spelregels - Regels Voor Blackjack | TOTO*, z.d.; *Blackjack: Uitleg & Spelregels*, 2025; Spelregels.Eu, 2020; Van Geest, 2014).

### 1.1.2 Het spel
Zodra de kaarten gedeeld heeft de speler twee keuzes: 'hit' en 'stick'. Als de speler kiest voor 'hit' dan krijgt de speler een nieuwe kaart, als de speler kiest voor 'stick' dan is het spel voor de speler afgelopen. In het casino zijn er ook de opties 'split', 'double down', 'surrender', echter zijn deze niet in de gymnasium library versie van Blackjack (*Blackjack Spelregels - Regels Voor Blackjack | TOTO*, z.d.; *Blackjack: Uitleg & Spelregels*, 2025; Spelregels.Eu, 2020; Van Geest, 2014). Wij hebben ervoor gekozen om de optie tot 'double down' ook in het spel te programmeren, dit is verwerkt in een inzet bewuste strategie.

### 1.1.3 De dealer
Als de speler klaar is met zijn spel, gaat de dealer zijn spel spelen. Hiervoor zijn twee vaste regels:

1. Als de bank minder dan 17 punten heeft moet er 'hit' gespeeld worden.
2. Als de bank 17 punten of meer heeft moet er 'stand' gespeeld worden.

### 1.1.4 Uitbetaling
Als het spel is afgelopen volgt de volgende uitbetaling (*Blackjack Spelregels - Regels Voor Blackjack | TOTO*, z.d.; *Blackjack: Uitleg & Spelregels*, 2025; Spelregels.Eu, 2020; Van Geest, 2014):

- Winst speler zonder Blackjack (exact 21 punten): 2x de inzet.
- Gelijkspel: inzet blijft staan voor de volgende ronde.
- Verlies: verlies inzet.
- Winst met Blackjack: 2.5x de inzet.

### 1.1.5 Waardes van de kaarten
In het Blackjack worden de volgende kaartwaardes gebruikt (*Blackjack Spelregels - Regels Voor Blackjack | TOTO*, z.d.; *Blackjack: Uitleg & Spelregels*, 2025; Spelregels.Eu, 2020; Van Geest, 2014):

- 2 t/m 10: eigen waarde.
- Aas: 1 of 11 punten (welke de som dichterbij de 21 krijgt).
- Boer, Vrouw en Heer: 10 punten.

## 1.2 Probleem

Het probleem dat centraal staat in dit onderzoek is het inherente gebrek aan winstkansen in het spel Blackjack. Hoewel Blackjack vaak wordt gezien als een spel waar vaardigheid een rol kan spelen, is de werkelijkheid dat het huisvoordeel (house edge) altijd in het voordeel van de casino ligt. Dit betekent dat, ongeacht de strategie die een speler hanteert, de kans op verlies op lange termijn groter is dan de kans op winst. Het huisvoordeel ontstaat doordat de dealer volgens vaste regels speelt en doordat bepaalde uitbetalingen, zoals bij een Blackjack, niet volledig compenseren voor de statistische nadelen.

Het onderzoek richt zich daarom niet op het vinden van een manier om te winnen, maar op het identificeren van strategieën die het verlies minimaliseren. Door middel van simulaties worden verschillende beslisregels geëvalueerd, zoals wanneer te 'hitten' of te 'sticken', en hoe deze keuzes invloed hebben op het gemiddelde verlies per ronde. Het doel is om inzichten te verkrijgen in welke strategieën leiden tot het minste verlies, rekening houdend met factoren zoals de huidige handwaarde, de zichtbare kaart van de dealer en de beschikbare inzet. Dit onderzoek draagt bij aan een beter begrip van kansspelen en kan dienen als basis voor verdere studies naar optimale besluitvorming in onzekere omgevingen.

# 2 Methodologie

In dit hoofdstuk wordt de methodologie van het onderzoek beschreven. Hierbij ligt de focus op de verschillende strategieën die zijn geïmplementeerd voor het spelen van blackjack, evenals de keuzes die zijn gemaakt bij het ontwerpen en uitvoeren van de simulaties. Het doel is om inzicht te krijgen in hoe verschillende beslisregels invloed hebben op het verloop van het spel en het uiteindelijke resultaat.

Naast de speelstrategie is ook een inzetstrategie meegenomen in het model. In elke ronde wordt de inzet bepaald als een vast percentage van het huidige aantal tokens van de speler, namelijk 25%. Dit betekent dat de inzet dynamisch verandert gedurende de simulatie: bij winst neemt de inzet toe, terwijl bij verlies de inzet juist afneemt. Deze aanpak zorgt ervoor dat de resultaten niet alleen afhankelijk zijn van de gekozen strategie, maar ook van hoe het aantal tokens dat beschikbaar is verandert over de tijd. 

De strategieën zijn geïmplementeerd als functies die, op basis van de huidige spelsituatie, een actie en bijbehorende inzet teruggeven. Deze functies worden herhaaldelijk aangeroepen binnen een simulatie over een groot aantal rondes, waarbij na elke ronde het aantal tokens wordt aangepast op basis van winst of verlies. Op deze manier kan de prestatie van elke strategie over de lange termijn worden geanalyseerd en met elkaar worden vergeleken.

## 2.1 Baseline strategie (random)

### Hoe werkt de strategie?
De baseline strategie voor blackjack werkt volledig willekeurig. Bij elke beurt kiest de speler met behulp van `random.choice` of hij een extra kaart neemt ("hit") of stopt ("stick"). Deze beslissing is niet gebaseerd op de waarde van de kaarten van de speler of de zichtbare kaart van de dealer, maar is dus puur random. Om deze reden vormt deze strategie de baseline om de andere strategieën mee te vergelijken. 

### Wanneer is dit goed + voordelen
Deze strategie is vooral geschikt als referentiepunt bij het vergelijken van de verschillende strategieën. Doordat er geen logica of voorkennis nodig is, is deze aanpak makkelijk te implementeren en te begrijpen. Dit maakt het een nuttige baseline om te laten zien hoeveel invloed een meer doordachte strategie kan hebben op de resultaten. Daarnaast kan de strategie op korte termijn soms positieve uitkomsten geven door puur geluk.

### Wanneer is dit slecht + nadelen
De strategie houdt geen rekening met de beste beslissingen binnen blackjack, waardoor de speler vaak niet de meest optimale keuzes maakt. Dit leidt ertoe dat de kans op winst aanzienlijk lager is dan bij strategieën die wel gebruik maken van spelinzicht. Op lange termijn zal deze strategie daarom vrijwel altijd slechter presteren dan andere strategieën. 

## 2.2 Volgens het boekje

### Hoe werkt de strategie?
De "boekje" strategie is gebaseerd op eenvoudige basisregels uit klassiek blackjack, waarbij de beslissingen van de speler afhangen van zowel de som van de kaarten van de speler als de zichtbare kaart van de dealer. Deze strategie maakt onderscheid tussen harde handen (zonder aas) en zachte handen (met aas), omdat een aas flexibel als 1 of 11 kan tellen.

Bij harde handen wordt er gewerkt met globale grenzen: tegen lage dealerkaarten (2 t/m 6) stopt de speler eerder, terwijl tegen hogere dealerkaarten (7 t/m aas) wordt langer doorgegaan met kaarten nemen. Bij zachte handen wordt er gekeken naar de aanwezigheid van lage kaarten naast de aas. Als de hand relatief zwak is, wordt er vaker een kaart genomen, terwijl bij sterkere combinaties eerder wordt gestopt. De strategie gebruikt alleen de "hit" en "stick" acties, double down en split zijn hierin niet meegenomen.

### Wanneer is dit goed + voordelen
Deze strategie is duidelijke effectiever dan een willekeurige aanpak, omdat beslissingen niet meer op toeval gebaseerd zijn, maar op informatie uit het spel. Er wordt rekening gehouden met zowel de zichtbare kaart van de dealer als het type hand van de speler (hard of zacht), wat belangrijke factoren zijn binnen blackjack. 

Daarnaast zorgt deze aanpak voor meer consistente resultaten op lange termijn, doordat systematisch betere keuzes worden gemaakt, wordt het huisvoordeel gedeeltelijk verkleind en neemt de kans op extreme verliezen af ten opzichte van willekeurig spelen.

### Wanneer is dit slecht + nadelen
Deze strategie houdt geen rekening met alle mogelijke acties binnen blackjack. In het bijozonder ontbreken double down en split, terwijl dit juist twee strategieën zijn die winstgevender zijn. Bijvoorbeeld bij een totaaal van 11 tegen een lage dealerkaart is verdubbelen vaak statistisch gezien voordelig, maar deze strategie kiest hier gewoon voor een hit. Hierdoor laat de speler dus winst liggen op lange termijn. 

Dit nadeel komt voort uit de beperkingen van de gebruikte Atari-omgeving, waarin deze acties niet standaard beschikbaar zijn. Hierdoor is bewust gekozen voor de vereenvoudigde versie van het boekje strategie die alleen gebruik maakt van hit en stick. 

Daarnaast zijn de beslisregels vrij globaal. Er wordt voornamelijk gekeken naar de totale waarde van de handen. Er wordt niet gekeken naar hoevaak bepaalde kaarten al zijn voorgekomen in een spel. Op het moment dat er veel lage kaarten zijn voorgekomen in eerdere rondes, is de kans hoger dan er bijvoorbeeld nu hoge kaarten komen. Dit zorgt ervoor het soms slim kan zijn om toch tegen de regels van het boekje in te spelen door kaarten te tellen. 

## 2.3 Inverse van het boekje

### Hoe werkt de strategie?
De inverse strategie is gebaseerd op de hiervoor beschreven "boekje" strategie, maar keert de beslisregels om. Waar de oorspronkelijke situatie kiest voor hit in situaties die statistisch gezien gunstig zijn, kiest deze strategie juiste voor stick en andersom.

Net als bij de originele strategie wordt er onderscheid gemaakt tussen harden handen (zonder aas) en zachte handen (met aas). De beslissingen wordt opnieuw gebaseerd op de som van de kaarten van de speler en de zichtbare kaart van de dealer, maar de acties zijn systematisch omgedraaid.

Een uitzondering hierop is de hand met een waarde van 21. In dat geval wordt er nog steeds gekozen voor stick, omdat het nemen van een extra kaart altijd zou leiden tot verlies. 

### Wanneer is dit goed + voordelen
De inverse strategie is niet bedoeld om winst te maken, maar juist om als vergelijkingsmateriaal te dienen. Door de resultaten van de inverse strategie te vergelijken met die van de normale strategie, kan worden aangetoond hoeveel invloed correcte beslissingen hebben op het verloop van het spel en het uiteindelijk aantal tokens. Het verschil in prestaties tussen beide strategieën geeft daarmee inzicht in de effectiviteit van het boekje. 


### Wanneer is dit slecht + nadelen
De inverse strategie is bewust suboptimaal en leidt in de meeste gevallen tot aanzienlijk slechtere resultaaten. Doordat goede beslissingen systematisch worden omgedraaid, wordt de kans op verlies vergroot en het huisvoordeel juist verwerkt in plaats van verkleind. Hierdoor zal de speler op lange termijn vrijwel altijd tokens verliezen. 

Net als bij de "boekje" strategie wordt er geen gebruik gemaakt van double down en split, wat de strategie minder realistisch maakt dan deze in werkelijkheid zou zijn. 

## 2.4 Hit t/m 17

### Hoe werkt de strategie?
De "Hit tot 17" strategie is een aanpak met een vaste grens. De speler neemt altijd een kaart (hit) zolang het totaal 17 of lager is, en stopt (stick) pas bij 18 of hoger (of bij exact 21). In tegenstelling tot de "boekje" strategie negeert deze methode de kaart van de dealer en het verschil tussen harde en zachte handen volledig. De speler kopieert hiermee simpelweg de standaardregel die in veel casino's voor de dealer zelf geldt.

### Wanneer is dit goed + voordelen
Het grootste voordeel is de eenvoud en consistentie. Emotionele beslissingen worden geëlimineerd omdat de regel altijd hetzelfde blijft. De strategie dwingt de speler om een competitief totaal na te streven en niet te vroeg te stoppen met een te lage score. Hierdoor presteert deze methode stabieler en aanzienlijk beter dan een willekeurige aanpak, waardoor het huisvoordeel tot een constant niveau wordt beperkt.

### Wanneer is dit slecht + nadelen
Het grootste nadeel is dat deze strategie blind is voor de situatie op tafel. Als de dealer een zwakke kaart heeft (zoals een 6), neemt de speler bij een eigen totaal van 15 alsnog het risico om "bust" te gaan, wat statistisch onnodig is (groter risico bij de speler dan de dealer). Daarnaast ontbreken winstgevende acties zoals double down en split. Door vast te houden aan de grens van 17, is de kans op verlies door een kapot gekochte hand groter dan bij een strategie die wel rekening houdt met de dealerkaart.

## 2.5 Stand vanaf 12

### Hoe werkt de strategie?
De "Stand op 12" strategie is een defensieve en voorzichtige methode. De speler neemt alleen een kaart (hit) bij een totaal lager dan 12. Zodra de hand een waarde van 12 of hoger bereikt, stopt de speler direct (stick). De achterliggende gedachte is simpel: vanaf 12 punten bestaat de kans dat je met één extra kaart (een 10 of plaatje) direct "bust" gaat. De strategie negeert de dealerkaart volledig en richt zich puur op het vermijden van een eigen fout.

### Wanneer is dit goed + voordelen
Het unieke voordeel van deze aanpak is dat de speler nooit door eigen toedoen verliest. Omdat je stopt voordat je jezelf dood kunt kopen, dwing je de dealer om jou te verslaan. Je wint elke ronde waarin de dealer zich "dood koopt" (bust), ongeacht hoe laag jouw eigen score is. Dit maakt de strategie zeer consistent en zorgt ervoor dat je langer met je tokens kunt doen, omdat je nooit het slachtoffer wordt van een ongelukkige extra kaart.

### Wanneer is dit slecht + nadelen
Het grote nadeel is dat je vaak eindigt met een zeer zwakke hand. Een score van 12 of 13 is in de meeste gevallen niet genoeg om te winnen als de dealer een geldige hand maakt. Omdat de strategie geen rekening houdt met de kaart van de dealer, toon je geen enkel initiatief om een betere hand te bouwen tegenover een sterke dealerkaart. Hierdoor lever je op de lange termijn veel winstkansen in; de dealer wint namelijk vaker met een gemiddelde hand dan dat hij zichzelf dood koopt.

## 2.6 Hit altijd

### Hoe werkt de strategie?
De "Hit altijd"-strategie is de meest agressieve en simpelste strategie: de speler vraagt altijd een extra kaart aan, ongeacht zijn eigen kaartensom, de zichtbare kaart van de dealer, of welke andere informatie dan ook. De enige uitzondering is wanneer de speler al een som van 21 heeft, dan wordt er gestopt om het punt niet te riskeren.

### Wanneer is dit goed + voordelen
De strategie is alleen voordelig aan het begin van een ronde, wanneer de som van de speler laag is (bijvoorbeeld 4–10). In die gevallen is de kans op busten (boven 21 gaan) klein, en is een extra kaart vrijwel altijd gunstig. In dit venster gedraagt de strategie zich identiek aan rationele strategieën. Bovendien is de strategie eenvoudig te implementeren zonder enige kennis of beslissingslogica, en kan het in zeldzame gevallen leiden tot een exacte 21 door meerdere kleine kaarten te trekken.

### Wanneer is dit slecht + nadelen
Zodra de speler een som heeft van 12 of hoger, neemt het bustrisico snel toe. Bij een som van 21 zou normaal stick de juiste keuze zijn, maar de code stopt pas bij exact 21. Bij 20, 19, of 18 blijft de speler onnodig kaarten trekken en bust hij in de meeste gevallen. Dit maakt de strategie structureel verliesgevend bij hogere sommen. Bovendien negeert de strategie volledig wanneer het verstandiger is om te stoppen, waardoor de speler structureel over de 21 gaat, en wordt geen gebruik gemaakt van informatie zoals de zichtbare kaart van de dealer of het bewustzijn van een soft hand (met Aas als 11). De strategie is zeer weinig robuust en presteert consistent slecht, ongeacht de omstandigheden, met een laag winstpercentage dat niet verbetert bij meer rondes.

## 2.7 Inzet aware hitting inclusief double down strategie

### Hoe werkt de strategie?
De inzet aware strategie is een directe uitbreiding op de "boekje" strategie (zie 2.2) en voegt een extra laag van besluitvorming toe: het verdubbelen van de inzet op statistisch gunstige momenten, ook wel "double down" genoemd. Net als bij de boekje strategie wordt er onderscheid gemaakt tussen harde handen (zonder bruikbare aas) en zachte handen (met een aas die als 11 telt), waarbij de beslissingen afhangen van zowel de eigen kaartensom als de zichtbare kaart van de dealer.

De double down beslissing wordt alleen overwogen als de speler precies twee kaarten heeft. Bij een harde hand wordt er verdubbeld bij een som van 11 tegen een dealerkaart tot en met 10, bij een som van 10 tegen een dealerkaart tot en met 9, en bij een som van 9 tegen een dealerkaart van 3 tot en met 6. Bij een zachte hand wordt verdubbeld bij een soft 17 tegen een dealerkaart van 3 tot en met 6, bij een soft 15 of 16 tegen een dealerkaart van 4 tot en met 6, en bij een soft 13 of 14 tegen een dealerkaart van 5 of 6. In alle overige gevallen blijft de inzet ongewijzigd.

Na de inzetbeslissing volgt de hit/stick beslissing, die grotendeels identiek is aan die van de boekje strategie: bij een zachte hand wordt er gehit zolang de som 17 of lager is, bij een harde hand wordt er gestopt vanaf 17, en bij 12 tot en met 16 wordt er gestopt als de dealer een lage kaart toont (2 tot en met 6) en gehit als de dealer een hoge kaart toont (7 of hoger). Bij een som van 21 wordt er altijd gestopt, en bij een som onder de 12 wordt altijd een extra kaart genomen.

### Wanneer is dit goed + voordelen
Deze strategie is het meest effectief in situaties waar de kansen duidelijk in het voordeel van de speler liggen. Het double down mechanisme is bijzonder krachtig bij een harde som van 11 of 10: de kans op een eindsom van 20 of 21 is in die gevallen hoog, en door de inzet te verdubbelen maximaliseert de speler het rendement op die gunstige momenten. Doordat de hit/stick beslissingen gebaseerd zijn op zowel de eigen hand als de zichtbare kaart van de dealer, zijn de keuzes statistisch onderbouwd en volgt de strategie bewezen blackjack-principes.

Onderscheid tussen harde/zachte handen maakt de aanpak strategisch rijker dan een simpele drempelwaarde. Bij een groot aantal gespeelde rondes zal de strategie naar verwachting beter presteren dan eenvoudigere varianten, doordat de verwachte waarde per ronde hoger ligt door het opportunistisch inzetten van de double down.

### Wanneer is dit slecht + nadelen
Het double down mechanisme vergroot niet alleen de potentiële winst, maar ook het verlies bij een slechte beurt. Als de inzet verdubbeld wordt maar de ronde toch verloren gaat, is de schade twee keer zo groot. In combinatie met de vaste inzetregel van 25% van de resterende tokens kan één mislukte double down een disproportioneel grote impact hebben, zeker wanneer de tokenpot al klein is.

Daarnaast slaagt de strategie alleen als je zachte handen goed identificeert. Zie je een lastige hand over het hoofd, dan haal je niet het maximale uit je spel. De hogere complexiteit leidt bovendien tot een grotere variantie in de resultaten: bij een kleine sample zoals 100 rondes domineert het toeval, waardoor de strategie in de praktijk soms slechter lijkt te presteren dan eenvoudigere alternatieven. De robuustheid van de strategie neemt pas duidelijk toe naarmate er meer rondes worden gespeeld en het statistisch voordeel van de double down zichtbaarder wordt.

# 3 Resultaten

## 3.0 Inladen libraries

In [ ]:
import gymnasium as gym
import matplotlib.pyplot as plt
import pandas as pd

import random

pd.set_option('display.float_format', lambda x: '%.2f' % x)

env = gym.make("Blackjack-v1")

sample_size = 100
tokens = 1000 # het aantal tokens om in te zetten tijdens het spelen van blackjack
inzet_percentage = 0.25 # het percentage van de tokens dat wordt ingezet bij elke hand

## 3.1 Opzetten spel

In [14]:
def blackjack(keuze_strategie, inzet):  
    obs, info = env.reset() 
    terminated = False
    truncated = False
    
    while not (terminated or truncated):
        speler_som = obs[0] 
        dealer_kaarten_zichtbaar = [env.unwrapped.dealer[0]]
        speler_kaarten = env.unwrapped.player
        
        keuze, inzet = keuze_strategie(speler_som, dealer_kaarten_zichtbaar, speler_kaarten, inzet) 
        
        if keuze == 'hit': 
            action = 1 
        elif keuze == 'stick':
            action = 0 
            
        obs, reward, terminated, truncated, info = env.step(action) 
        
    if reward > 0:
        if len(speler_kaarten) == 2 and 1 in speler_kaarten and 10 in speler_kaarten:  
            winst = inzet * 1.5 
            resultaat = 'Blackjack'
        else:
            winst = inzet 
            resultaat = 'Gewonnen'
    elif reward < 0:
        winst = -inzet 
        resultaat = 'Verloren'
    else:
        winst = 0 
        resultaat = 'Gelijk'
        
    return winst, resultaat

## 3.2 Strategieën uitwerken en uitvoeren

In [ ]:
def baseline_strat(speler_som, dealer_kaarten_zichtbaar, speler_kaarten, inzet):
    return random.choice(['hit', 'stick']), inzet

baseline_results = [] 
baseline_tokens = tokens 
for _ in range(sample_size):
    inzet = baseline_tokens * inzet_percentage
    resultaat = blackjack(baseline_strat, inzet)
    baseline_tokens += resultaat[0]
    baseline_results.append(resultaat[1])
    
baseline_tokens = baseline_tokens if baseline_tokens >= 0.01 else 0

print(f"Na {sample_size} rondes met de baseline strategie heeft de speler {baseline_tokens} tokens over.")

Na 100 rondes met de baseline strategie heeft de speler 0 tokens over.


In [ ]:
def strat1_boekje(speler_som, dealer_kaarten_zichtbaar, speler_kaarten, inzet):
    if speler_som == 21:
        return 'stick', inzet 
    elif 1 not in speler_kaarten:
        if dealer_kaarten_zichtbaar[0] in [2, 3]:
            if speler_som <= 12:
                return 'hit', inzet
            else:
                return 'stick', inzet
        elif dealer_kaarten_zichtbaar[0] in [4, 5, 6]:
            if speler_som <= 11:
                return 'hit', inzet
            else:
                return 'stick', inzet
        elif dealer_kaarten_zichtbaar[0] in [7, 8, 9, 10, 1]:
            if speler_som <= 16:
                return 'hit', inzet
            else:
                return 'stick', inzet
    else:
        if dealer_kaarten_zichtbaar[0] in [2, 3, 4, 5, 6, 7, 8]: 
            if any(kaart in speler_kaarten for kaart in [2, 3, 4, 5, 6]):
                return 'hit', inzet
            else:
                return 'stick', inzet
        elif dealer_kaarten_zichtbaar[0] in [9, 10, 1]:
            if any(kaart in speler_kaarten for kaart in [2, 3, 4, 5, 6, 7]):
                return 'hit', inzet
            else:
                return 'stick', inzet
        
    
strat1_results = []
strat1_tokens = tokens 
for _ in range(sample_size):
    inzet = strat1_tokens * inzet_percentage
    resultaat = blackjack(strat1_boekje, inzet)
    strat1_tokens += resultaat[0]
    strat1_results.append(resultaat[1]) 
    
strat1_tokens = strat1_tokens if strat1_tokens >= 0.01 else 0

print(f"Na {sample_size} rondes met de boekje strategie heeft de speler {strat1_tokens} tokens over.")

Na 100 rondes met de boekje strategie heeft de speler 62.04451636093149 tokens over.


In [ ]:
def strat2_boekje_invers(speler_som, dealer_kaarten_zichtbaar, speler_kaarten, inzet):
    if speler_som == 21:  
        return 'stick', inzet
    elif 1 not in speler_kaarten: 
        if dealer_kaarten_zichtbaar[0] in [2, 3]: 
            if speler_som <= 12:
                return 'stick', inzet
            else:
                return 'hit', inzet
        elif dealer_kaarten_zichtbaar[0] in [4, 5, 6]: 
            if speler_som <= 11:
                return 'stick', inzet
            else:
                return 'hit', inzet
        elif dealer_kaarten_zichtbaar[0] in [7, 8, 9, 10, 1]:
            if speler_som <= 16:
                return 'stick', inzet
            else:
                return 'hit', inzet
    else:
        if dealer_kaarten_zichtbaar[0] in [2, 3, 4, 5, 6, 7, 8]: 
            if any(kaart in speler_kaarten for kaart in [2, 3, 4, 5, 6]):
                return 'stick', inzet
            else:
                return 'hit', inzet
        elif dealer_kaarten_zichtbaar[0] in [9, 10, 1]: 
            if any(kaart in speler_kaarten for kaart in [2, 3, 4, 5, 6, 7]):
                return 'stick', inzet
            else:
                return 'hit', inzet
        
    
strat2_results = []
strat2_tokens = tokens 
for _ in range(sample_size):
    inzet = strat2_tokens * inzet_percentage
    resultaat = blackjack(strat2_boekje_invers, inzet)
    strat2_tokens += resultaat[0]
    strat2_results.append(resultaat[1])

strat2_tokens = strat2_tokens if strat2_tokens >= 0.01 else 0

print(f"Na {sample_size} rondes met de inverse strategie van het boekje heeft de speler {strat2_tokens} tokens over.")

Na 100 rondes met de inverse strategie van het boekje heeft de speler 0 tokens over.


In [ ]:
def strat3_hit_17(speler_som, dealer_kaarten_zichtbaar, speler_kaarten, inzet):
    if speler_som == 21: 
        return 'stick', inzet
    elif speler_som <= 17:
        return 'hit', inzet
    else:
        return 'stick', inzet
        
    
strat3_results = []
strat3_tokens = tokens 
for _ in range(sample_size):
    inzet = strat3_tokens * inzet_percentage
    resultaat = blackjack(strat3_hit_17, inzet)
    strat3_tokens += resultaat[0]
    strat3_results.append(resultaat[1])

strat3_tokens = strat3_tokens if strat3_tokens >= 0.01 else 0

print(f"Na {sample_size} rondes met de hit tot 17 strategie heeft de speler {strat3_tokens} tokens over.")

Na 100 rondes met de hit tot 17 strategie heeft de speler 36.09862770090559 tokens over.


In [ ]:
def strat4_stand_12(speler_som, dealer_kaarten_zichtbaar, speler_kaarten, inzet):
    if speler_som == 21:  
        return 'stick', inzet
    elif speler_som >= 12:
        return 'stick', inzet
    else:
        return 'hit', inzet
        
    
strat4_results = []
strat4_tokens = tokens 
for _ in range(sample_size):
    inzet = strat4_tokens * inzet_percentage
    resultaat = blackjack(strat4_stand_12, inzet)
    strat4_tokens += resultaat[0]
    strat4_results.append(resultaat[1])

strat4_tokens = strat4_tokens if strat4_tokens >= 0.01 else 0

print(f"Na {sample_size} rondes met de stand op 12 strategie heeft de speler {strat4_tokens} tokens over.")

Na 100 rondes met de stand op 12 strategie heeft de speler 34.94347161447666 tokens over.


In [ ]:
def strat5_dealer_weakness(speler_som, dealer_kaarten_zichtbaar, speler_kaarten, inzet):
    if speler_som == 21:  
        return 'stick', inzet
    elif dealer_kaarten_zichtbaar[0] in [2, 3, 4, 5, 6]: 
        return 'stick', inzet
    elif dealer_kaarten_zichtbaar[0] in [7, 8, 9, 10, 1]:
        if speler_som <= 21:
            return 'hit', inzet
        else:
            return 'stick', inzet
        
    
strat5_results = []
strat5_tokens = tokens 
for _ in range(sample_size):
    inzet = strat5_tokens * inzet_percentage
    resultaat = blackjack(strat5_dealer_weakness, inzet)
    strat5_tokens += resultaat[0]
    strat5_results.append(resultaat[1])

strat5_tokens = strat5_tokens if strat5_tokens >= 0.01 else 0

print(f"Na {sample_size} rondes met de dealer weakness strategie heeft de speler {strat5_tokens} tokens over.")

Na 100 rondes met de dealer weakness strategie heeft de speler 0 tokens over.


In [ ]:
def strat6_always_hit(speler_som, dealer_kaarten_zichtbaar, speler_kaarten, inzet):
    if speler_som != 21:  
        return 'hit', inzet
    else:
        return 'stick', inzet
        
    
strat6_results = []
strat6_tokens = tokens 
for _ in range(sample_size):
    inzet = strat6_tokens * inzet_percentage
    resultaat = blackjack(strat6_always_hit, inzet)
    strat6_tokens += resultaat[0] 
    strat6_results.append(resultaat[1])

strat6_tokens = strat6_tokens if strat6_tokens >= 0.01 else 0

print(f"Na {sample_size} rondes met de always hit strategie heeft de speler {strat6_tokens} tokens over.")

Na 100 rondes met de always hit strategie heeft de speler 0 tokens over.


In [ ]:
def strat7_inzet_aware(speler_som, dealer_kaarten_zichtbaar, speler_kaarten, inzet):
    dealer_kaart = dealer_kaarten_zichtbaar[0] 
    is_soft = 1 in speler_kaarten and (speler_som + 10 <= 21)
    aantal_kaarten = len(speler_kaarten) 

    if aantal_kaarten == 2:
        if not is_soft:
            if speler_som == 11 and dealer_kaart <= 10: 
                nieuwe_inzet = inzet * 2 
            elif speler_som == 10 and dealer_kaart <= 9:
                nieuwe_inzet = inzet * 2 
            elif speler_som == 9 and dealer_kaart in [3, 4, 5, 6]:
                nieuwe_inzet = inzet * 2 
            else:
                nieuwe_inzet = inzet
        else:
            if speler_som == 17 and dealer_kaart in [3, 4, 5, 6]:
                nieuwe_inzet = inzet * 2 
            elif (speler_som == 16 or speler_som == 15) and dealer_kaart in [4, 5, 6]:
                nieuwe_inzet = inzet * 2 
            elif (speler_som == 14 or speler_som == 13) and dealer_kaart in [5, 6]:
                nieuwe_inzet = inzet * 2 
            else:
                nieuwe_inzet = inzet
    else:
        nieuwe_inzet = inzet
    
    if speler_som >= 21:
        return 'stick', nieuwe_inzet 

    if is_soft: 
        if speler_som <= 17: 
            return 'hit', nieuwe_inzet
        else:
            return 'stick', nieuwe_inzet

    if speler_som >= 17: 
        return 'stick', nieuwe_inzet
    elif 12 <= speler_som <= 16: 
        if dealer_kaart <= 6:
            return 'stick', nieuwe_inzet
        else:
            return 'hit', nieuwe_inzet
    else:
        return 'hit', nieuwe_inzet
        
strat7_results = []
strat7_tokens = tokens 
for _ in range(sample_size):
    inzet = strat7_tokens * inzet_percentage
    resultaat = blackjack(strat7_inzet_aware, inzet)
    strat7_tokens += resultaat[0] 
    strat7_results.append(resultaat[1])

strat7_tokens = strat7_tokens if strat7_tokens >= 0.01 else 0

print(f"Na {sample_size} rondes met de inzet aware strategie heeft de speler {strat7_tokens} tokens over.")

Na 100 rondes met de inzet aware strategie heeft de speler 24.52819620678791 tokens over.


## 3.3 Strategieën vergelijken

In [23]:
df = pd.DataFrame({
    'Strategie': ['Random (Baseline)', 'Boekje', 'Boekje Invers', 'Hit tot 17', 'Stand vanaf 12', 'Dealer Weakness', 'Always Hit', 'Inzet Aware'],
    'Blackjack (%)': [round(baseline_results.count('Blackjack') / sample_size * 100, 1), round(strat1_results.count('Blackjack') / sample_size * 100, 1), round(strat2_results.count('Blackjack') / sample_size * 100, 1), round(strat3_results.count('Blackjack') / sample_size * 100, 1), round(strat4_results.count('Blackjack') / sample_size * 100, 1), round(strat5_results.count('Blackjack') / sample_size * 100, 1), round(strat6_results.count('Blackjack') / sample_size * 100, 1), round(strat7_results.count('Blackjack') / sample_size * 100, 1)],
    'Gewonnen (%)': [round((baseline_results.count('Gewonnen') + baseline_results.count('Blackjack')) / sample_size * 100, 1), round((strat1_results.count('Gewonnen') + strat1_results.count('Blackjack')) / sample_size * 100, 1), round((strat2_results.count('Gewonnen') + strat2_results.count('Blackjack')) / sample_size * 100, 1), round((strat3_results.count('Gewonnen') + strat3_results.count('Blackjack')) / sample_size * 100, 1), round((strat4_results.count('Gewonnen') + strat4_results.count('Blackjack')) / sample_size * 100, 1), round((strat5_results.count('Gewonnen') + strat5_results.count('Blackjack')) / sample_size * 100, 1), round((strat6_results.count('Gewonnen') + strat6_results.count('Blackjack')) / sample_size * 100, 1), round((strat7_results.count('Gewonnen') + strat7_results.count('Blackjack')) / sample_size * 100, 1)],
    'Verloren (%)': [round(baseline_results.count('Verloren') / sample_size * 100, 1), round(strat1_results.count('Verloren') / sample_size * 100, 1), round(strat2_results.count('Verloren') / sample_size * 100, 1), round(strat3_results.count('Verloren') / sample_size * 100, 1), round(strat4_results.count('Verloren') / sample_size * 100, 1), round(strat5_results.count('Verloren') / sample_size * 100, 1), round(strat6_results.count('Verloren') / sample_size * 100, 1), round(strat7_results.count('Verloren') / sample_size * 100, 1)],
    'Gelijk (%)': [round(baseline_results.count('Gelijk') / sample_size * 100, 1), round(strat1_results.count('Gelijk') / sample_size * 100, 1), round(strat2_results.count('Gelijk') / sample_size * 100, 1), round(strat3_results.count('Gelijk') / sample_size * 100, 1), round(strat4_results.count('Gelijk') / sample_size * 100, 1), round(strat5_results.count('Gelijk') / sample_size * 100, 1), round(strat6_results.count('Gelijk') / sample_size * 100, 1), round(strat7_results.count('Gelijk') / sample_size * 100, 1)],
    'Overgebleven Tokens': [round(baseline_tokens, 2), round(strat1_tokens, 2), round(strat2_tokens, 2), round(strat3_tokens, 2), round(strat4_tokens, 2), round(strat5_tokens, 2), round(strat6_tokens, 2), round(strat7_tokens, 2)],
    'Tokens verlies (%)': [round((tokens - baseline_tokens) / tokens * 100, 1) if baseline_tokens < tokens else 0, round((tokens - strat1_tokens) / tokens * 100, 1) if strat1_tokens < tokens else 0, round((tokens - strat2_tokens) / tokens * 100, 1) if strat2_tokens < tokens else 0, round((tokens - strat3_tokens) / tokens * 100, 1) if strat3_tokens < tokens else 0, round((tokens - strat4_tokens) / tokens * 100, 1) if strat4_tokens < tokens else 0, round((tokens - strat5_tokens) / tokens * 100, 1) if strat5_tokens < tokens else 0, round((tokens - strat6_tokens) / tokens * 100, 1) if strat6_tokens < tokens else 0, round((tokens - strat7_tokens) / tokens * 100, 1) if strat7_tokens < tokens else 0],
    'Tokens winst (%)': [round((baseline_tokens - tokens) / tokens * 100, 1) if baseline_tokens > tokens else 0, round((strat1_tokens - tokens) / tokens * 100, 1) if strat1_tokens > tokens else 0, round((strat2_tokens - tokens) / tokens * 100, 1) if strat2_tokens > tokens else 0, round((strat3_tokens - tokens) / tokens * 100, 1) if strat3_tokens > tokens else 0, round((strat4_tokens - tokens) / tokens * 100, 1) if strat4_tokens > tokens else 0, round((strat5_tokens - tokens) / tokens * 100, 1) if strat5_tokens > tokens else 0, round((strat6_tokens - tokens) / tokens * 100, 1) if strat6_tokens > tokens else 0, round((strat7_tokens - tokens) / tokens * 100, 1) if strat7_tokens > tokens else 0]
})

In [24]:
df

,Strategie,Blackjack (%),Gewonnen (%),Verloren (%),Gelijk (%),Overgebleven Tokens,Tokens verlies (%),Tokens winst (%)
0,Random (Baseline),2.00,30.00,65.00,5.00,0.00,100.00,0
1,Boekje,5.00,46.00,47.00,7.00,62.04,93.80,0
2,Boekje Invers,5.00,23.00,77.00,0.00,0.00,100.00,0
3,Hit tot 17,4.00,44.00,47.00,9.00,36.10,96.40,0
4,Stand vanaf 12,6.00,43.00,47.00,10.00,34.94,96.50,0
5,Dealer Weakness,1.00,31.00,66.00,3.00,0.00,100.00,0
6,Always Hit,2.00,15.00,84.00,1.00,0.00,100.00,0
7,Inzet Aware,4.00,40.00,47.00,13.00,24.53,97.50,0


## 3.4 Bevindingen

In dit paragraaf worden de resultaten van de simulaties gepresenteerd en geanalyseerd. Er zijn acht verschillende blackjackstrategieën getest in de `Blackjack-v1` omgeving van Gymnasium. Elke strategie werd uitgevoerd over een steekproef van 100 speelrondes (`sample_size = 100`), beginnend met een startkapitaal van 1000 tokens. Het inzetmodel was dynamisch: per ronde werd 25% van het huidige kapitaal ingezet, wat de impact van volatiliteit en opeenvolgende verliezen benadrukt.

De resultaten zijn gecategoriseerd in 'Gewonnen' (inclusief Blackjacks), 'Verloren' en 'Gelijkspel'. De simulaties zijn uitgevoerd met een vaste random seed (42) voor reproduceerbaarheid.

**Tabel 1: Simulatiescores per blackjackstrategie (100 rondes, startkapitaal 1000 tokens)**

| Strategie          | Eindkapitaal (Tokens) | Win/Blackjack (%) | Verloren (%) |
|--------------------|-----------------------|-------------------|--------------|
| Boekje (Basic Chart) | 62.04                | 46%              | 47%         |
| Hit tot 17         | 36.10                | 44%              | 47%         |
| Stand vanaf 12     | 34.94                | 43%              | 47%         |
| Inzet Aware        | 24.53                | 40%              | 47%         |
| Random (Baseline)  | 0.00                 | 30%              | 65%         |
| Boekje Invers      | 0.00                 | 23%              | 77%         |
| Dealer Weakness    | 0.00                 | 31%              | 66%         |
| Always Hit         | 0.00                 | 15%              | 84%         |

*Opmerking: De resterende percentages tot 100% betreffen gelijkspellen.*

**Belangrijkste inzichten:**
- **Kapitaalerosie:** Bijna alle strategieën leiden tot aanzienlijk verlies over 100 rondes, vanwege het huisvoordeel (house edge) in blackjack.
- **Overlevende strategieën:** Alleen de Boekje, Hit tot 17, Stand vanaf 12 en Inzet Aware strategieën voorkwamen volledig faillissement, al eindigden ze met een fractie van het startkapitaal. Hit tot 17 presteerde relatief het best in termen van risicobeperking.
- **Ondermaats presterende strategieën:** Random, Boekje Invers, Dealer Weakness en Always Hit resulteerden in volledig verlies van kapitaal.
- **Effect van dynamische inzet:** De Inzet Aware strategie verhoogt de variantie, wat theoretisch meer winstpotentieel biedt, maar in de praktijk het risico op snelle verliezen vergroot. Het eindigde dicht bij break-even in winratio, maar met aanzienlijk kapitaalverlies.

# 4 Conclusie

De simulaties tonen duidelijk aan dat er geen betrouwbare strategie bestaat om blackjack structureel te winnen in deze opstelling. Zelfs de meest optimale strategieën, zoals de Boekje-strategie of Hit tot 17, leiden tot significant kapitaalverlies bij een dynamische inzet van 25% per ronde. Dit bevestigt het statistische principe dat het huisvoordeel altijd in het voordeel van de casino werkt.

Bij een kleine steekproef van 100 rondes domineert variantie (geluk) de resultaten, waardoor geen strategie een positieve rendementscurve genereert. Inzet-adaptieve maatregelen, zoals in de Inzet Aware strategie, vergroten de volatiliteit maar resulteren niet in winst.

**Aanbeveling en disclaimer:** Dit onderzoek is puur academisch en analyseert risico's en statistische variantie. Het biedt geen model voor financieel gewin. De data bewijst dat het verslaan van het spel onder deze condities wiskundig onwaarschijnlijk is.

In de woorden van ons advies: "Gok niet."

# 5 Reflectie

Dit onderzoek naar verschillende strategieën in Blackjack heeft waardevolle inzichten opgeleverd over kansspelen, besluitvorming en simulatie. Als groep hebben we geleerd dat, ondanks de illusie van controle door strategieën, het huisvoordeel altijd dominant blijft. De implementatie van strategieën zoals het Boekje en Hit tot 17 toonde aan hoe eenvoudige regels kunnen leiden tot betere resultaten, maar nog steeds niet tot winst op lange termijn.

Een van de grootste uitdagingen was het werken met de Gymnasium Blackjack-omgeving, die beperkingen heeft zoals het ontbreken van double down en split in de basisversie. Dit dwong ons om creatieve aanpassingen te maken, zoals de Inzet Aware strategie, wat ons inzicht gaf in de complexiteit van spelregels. Bovendien bleek de dynamische inzet van 25% per ronde een grote impact te hebben op de volatiliteit, wat ons leerde over risicobeheer in simulaties.

Persoonlijk reflecterend, heeft dit project ons bewust gemaakt van de gevaren van gokken. De data toont duidelijk dat geen strategie veilig is, wat een belangrijke les is over verwachtingswaarde en variantie. Voor toekomstige onderzoeken zouden we graag meer strategieën testen, zoals kaart tellen, of de steekproef vergroten om variantie te verminderen. Ook zou het interessant zijn om reinforcement learning toe te passen voor adaptieve strategieën.

Al met al, dit project heeft ons niet alleen technische vaardigheden bijgebracht in Python en simulatie, maar ook een dieper begrip van statistiek en besluitvorming onder onzekerheid.

# Literatuurlijst

- *Blackjack spelregels - Regels voor Blackjack | TOTO.* (z.d.). Geraadpleegd op 12 maart 2026, van https://www.toto.nl/klantenservice/spelregels-voor-blackjack
- *Blackjack: uitleg & spelregels.* (2025, 4 november). Club BetCity. Geraadpleegd op 12 maart 2026, van https://club.betcity.nl/blackjack/regels
- Spelregels.Eu. (2020, 20 april). *Hoe speel je Blackjack.* spelregels.eu. Geraadpleegd op 12 maart 2026, van https://www.spelregels.eu/blackjack
- Van Geest, M. (2014). *Blackjack: regels en uitleg.* Meneer Casino. Geraadpleegd op 12 maart 2026, van https://meneercasino.com/online-casino-tips/blackjack-regels-en-uitleg